### PR TITLE
feat(rna): Add SignIn subcomponent

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add `SignIn` subcomponent

#### Issue #, if available
NA

#### Description of how you validated changes
Manually tested, unit tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-native/src/Authenticator/Defaults/SignIn/SignIn.tsx [hunk 1]
```diff
@@ -1,13 +1,95 @@
...
+  socialProviders,
+  toResetPassword,
+  toFederatedSignIn,
+  toSignUp,
+}) => {
+  return (
+    <View style={styles.container}>
+      {hideSignUp ? null : (
+        <Tabs style={styles.tabs}>
+          <Tab>Sign In</Tab>
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Defaults/SignIn/SignIn.tsx:62]: Now that Will's shared Authenticator text PR is in `rna/main`, can we use it for strings?

File: packages/react-native/src/Authenticator/Defaults/SignIn/__tests__/SignIn.spec.tsx [hunk 2]
```diff
@@ -1,22 +1,65 @@
...
-  it('renders as expected', () => {
-    const { toJSON, getByRole } = render(
-      <>
-        <SignIn {...props} />
-        <SignIn.Header />
-        <SignIn.Footer />
-        <SignIn.FormFields {...props} />
-      </>
-    );
+  it('primary component renders as expected', () => {
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Defaults/SignIn/__tests__/SignIn.spec.tsx:27]: nit: Can we remove `'primary component'` from each test description?

File: packages/react-native/src/Authenticator/Defaults/SignIn/__tests__/SignIn.spec.tsx [hunk 3]
```diff
@@ -1,22 +1,65 @@
...
+  });
+
+  it('primary component renders as expected with socialProviders', () => {
+    const { toJSON } = render(
+      <SignIn {...props} socialProviders={['amazon', 'facebook']} />
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('primary component renders as expected when socialProviders is undefined', () => {
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Defaults/SignIn/__tests__/SignIn.spec.tsx:59]: Isn't this case covered by the first unit test?

File: examples/react-native/storybook/stories/SignIn.stories.tsx [hunk 4]
```diff
@@ -1,11 +1,50 @@
...
+  isPending: false,
+  handleBlur: noop,
+  handleSubmit: (values: any) => {
+    console.log('Values', values);
+  },
+  handleChange: noop,
+  toFederatedSignIn: noop,
+  toResetPassword: noop,
+  toSignUp: noop,
+  error: null as unknown as string,
```
Comment (joebuono) [inline examples/react-native/storybook/stories/SignIn.stories.tsx:37]: nit: Can this be simplified?

Comment (calebpollman) [reply examples/react-native/storybook/stories/SignIn.stories.tsx:37]: How so?

File: packages/react-native/src/Authenticator/Defaults/SignIn/__tests__/SignIn.spec.tsx [hunk 5]
```diff
@@ -1,22 +1,65 @@
...
+  DefaultFormFields,
+  DefaultHeader,
+} from '../../../common';
+import SignIn from '../SignIn';
 
-const props = {} as any;
+const props = {
+  fields: [],
+  FormFields: DefaultFormFields,
+  Footer: DefaultFooter,
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Defaults/SignIn/__tests__/SignIn.spec.tsx:14]: Could we use `SignIn.Footer`?

**Review by joebuono — APPROVED** [review]

File: packages/react-native/src/Authenticator/Defaults/SignIn/SignIn.tsx [hunk 6]
```diff
@@ -1,13 +1,103 @@
...
+  FormFields,
+  Header,
+  hideSignUp,
+  isPending,
+  socialProviders,
+  toResetPassword,
+  toFederatedSignIn,
+  toSignUp,
+}) => {
+  const {
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Defaults/SignIn/SignIn.tsx:60]: nit: Would it be better to destructure these outside of `SignIn`?

**Review by ioanabrooks — APPROVED** [review]

**Pull request merged**